### PR TITLE
gym plugin with new structure

### PR DESCRIPTION
**Pull request by jordis-ai2**

### CODE REVIEW

**Comment by jordis-ai2** [comment]
I thought it would be easier for me to update to the new structure. We can merge it together with the rest of the restructuring (if you like).

**Review by Lucaweihs — COMMENTED** [review]
This is fantastic, so cool that we can do this now! One tiny comment but feel free to merge.

File: allenact_plugins/gym_plugin/gym_tasks.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        }
+
+
+class GymContinuousBox2DTask(GymTask):
+    """ Task for a continuous-control gym Box2D Env; it allows interfacing allenact with
+    gym tasks.
+    """
+
+    def class_action_names(cls, **kwargs) -> Tuple[str, ...]:
+        return ()
```
[the remark this answers is not in the document]
Comment (jordis-ai2) [reply allenact_plugins/gym_plugin/gym_tasks.py:79]: One problem that I see is that the action_space is only defined at the instance level, whereas class_action_names is at the class level. I'm not sure that I understand what you would like to get implemented. Can you develop it (obviously when you have some time?). Sorry for the delay...

[the remark this answers is not in the document]
Comment (jordis-ai2) [reply allenact_plugins/gym_plugin/gym_tasks.py]: For now, I'll merge with `tuple()` being returned by the task with non-categorical action space.

**Comment by jordis-ai2** [comment]
Merging with empty class names returned by the gym task with Box action space.

**Pull request merged**